### DISCUSSION TITLE
Deprecate STATE_STANDBY in media players

### DISCUSSION THREAD

**Discussion by Drafteed** [home-assistant/architecture#799] [Entity Models] [answered]
## Proposal

Deprecate `STATE_STANDBY` in media_player as a follow-up to https://github.com/home-assistant/core/pull/77941 and use `STATE_OFF`  with `available` property or `STATE_IDLE` instead.

I suggest using only the state `ON`, `OFF` (in combination `available` property) or `IDLE`, which means that the device is on and does not play anything to finally avoid confusion states.

## Current behavior

At this moment, 9 integrations use `STATE_STANDBY`, and each interprets it in its own way, that confuses both users and developers. Some integrations interprets this to be `off`, others interprets it to be `on` and `idle`.

- [Android TV](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/androidtv/media_player.py#L90) uses `STATE_STANDBY`, when the device is on but showing a screensaver. In this case, the state should be `STATE_IDLE`.
- [Apple TV](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/apple_tv/media_player.py#L177-L197) uses `STATE_STANDBY`, when the device is available but turned off. In this case, the state should be `STATE_OFF`, or `unavailable` if the device is not connected.
- [Google Assistant](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/google_assistant/trait.py#L2262-L2282) interprets `STATE_STANDBY` as on and idle.
- [HomeKit](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/homekit/type_media_players.py#L71-L76) interprets `STATE_STANDBY` as off.
- [Lookin](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/lookin/media_player.py#L112-L116) uses `STATE_STANDBY` instead of `STATE_OFF`.
- [Mediaroom](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/mediaroom/media_player.py#L160) uses `STATE_STANDBY` when device is on, but inactive. In this case, the state should `STATE_IDLE`.
- [PS4](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/ps4/media_player.py#L166-L203) uses `STATE_STANDBY` instead of `STATE_OFF`.
- [Roku](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/roku/media_player.py#L153-L176) uses `STATE_STANDBY` instead of `STATE_OFF`.
- [Universal](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/universal/media_player.py#L102-L112) gives priority to the standby state over the idle state.

Also [Wemo](https://github.com/home-assistant/core/blob/52c8c80f91283dd2665cd242d33131d98dfbcb17/homeassistant/components/wemo/switch.py#L106-L119) uses `STATE_STANDBY` in switch.

**Comment by frenck** [comment]
Besides some integrations using the wrong state, standby, off and idle are three completely different/distinctive states to me.

Why would standby need to go? And how would one differentiate between een device being off or in standby mode?

Standby isn't the same as idle, and idle and off aren't the same either.

So I am struggling understand the suggestion, which in my opinion, isn't explained in the above.

**Reply by Drafteed** [reply]
For me, as probably for most users, the standby mode means "available, but turned off". In this case, we can use the `available` property and the `STATE_OFF` specially created for this. However, as we can see, some may interpret this as "on, available and idle". In this case, we should use `STATE_IDLE` specially created for this. 

Generally, this state only creates confusion for both developers and users who have to handle this state in automations.

I also read the arguments of some people from here https://github.com/home-assistant/architecture/discussions/504 and concluded that standby state is necessary.

**Comment by thecode** [comment]
I agree, although hardware wise most TVs can be in standby mode or off (unpluged), most intergrations don't/can't diffrentiate between the two as in standby mode devices usually does not answer to requests. 
Also unless for a technical user, the media player is OFF in this state.

**Reply by Drafteed** [reply]
> in standby mode devices usually does not answer to requests.

For example, the API of my TV (Bravia) gives two states: standby and active. In standby mode of the TV, I can turn it on via request or get a list of sources or anything else. Active means that TV is on. If the device is not available, then it, accordingly, does not give anything to the API. In this case state should be `STATE_UNAVAILABLE`.

**Comment by elupus** [comment]
The distinction would really be if it is able to report state like volume, and adjust them. In off, this is not possible. Standby could allow that.

**Reply by Drafteed** [reply]
If we can change the volume of the device, is it turned off? I think it's in IDLE mode.

**Comment by Drafteed** [comment]
Even in this discussion, each of us interprets standby in his own way, someone considers it to be off, someone half on, or even idle. 

**To finally** avoid confusion, I suggest using only the state ON, OFF (in combination with available/unvailable) or IDLE, which means that the device is on and does not play anything.

**Comment by emontnemery** [comment, marked as answer]
I think this suggestion makes sense. There's no other platform where we distinguish between standby and off states.
- If a remote device is in a state where it's no longer connected to the network and can't be controlled remotely, it should be marked as unavailable.
- If a remote device is in a state where it first needs to be sent an on command before it will accept other commands, the command sequencing is for the integration to deal with

Most media player integrations do not make use of it, and the way in which it's used in very inconsistent as summarized by @Drafteed.

In the almost two years which have passed since this proposal was created, two additional integrations use the standby state (which is now called `MediaPlayerState.STANDBY`):
- [Cambridge Audio](https://github.com/home-assistant/core/blob/3b64db5f767aa2e632a53d8b33a72f6f08c89de0/homeassistant/components/cambridge_audio/media_player.py#L110-L111) uses STATE_STANDBY instead of STATE_OFF
- [Snapcast](https://github.com/home-assistant/core/blob/3b64db5f767aa2e632a53d8b33a72f6f08c89de0/homeassistant/components/snapcast/media_player.py#L346-L347)  uses STATE_STANDBY instead of STATE_OFF

**Reply by emontnemery** [reply]
This suggestion has been discussed and approved by the core team, let's go ahead with:
- Core implementation PR
- Updated [developer documentation](https://developers.home-assistant.io/docs/core/entity/media-player)
- Updated [user documentation](https://www.home-assistant.io/integrations/media_player/)
- A developer blog post

**Comment by parautenbach** [comment]
Perhaps picking on semantics, but I just wanted to add that off to me means not powered. I'm very used to many devices are in a sleeping state when in standby (the case where an Android TV uses standby when showing a screensaver seems odd – should be idle, in my view, which is on but doing nothing). Sleeping is different from being off, in my view. 

The clean-up looks like a good idea to me otherwise.

**Comment by jamie-0** [comment]
The decision has confused things futher for me if I’m honest.

Being off and on is a binary state. Why remove standby but keep the concept of idle when something can only be idle when it’s on and not off. Can something also be unavailable and on? 

In the case of the Cambridge audio a device needs to be specifically set to networked mode so when the device “powers off” but keeps the network stack on. To the user the device is powered off. There’s no concept of standby.  Hitting the power button is needed to turn it back on or you can use the networked functionality to do so.

So standby/available would be the normal state, just like a smart tv might be in standby and keeps its network connection on. 

A unit the has been configured in this way but has lost network connectivity would be standby/unavailable.

A unit that has been been deliberately turned off using the “eco” mode would be represented by off. The user has to physically  press the power button to turn it back on.

**Reply by jianyu-li** [reply]
There's a growing issue over on the Apple Tv side of things because the new state of OFF instead of IDLE prohibits us from tracking Tv playback state when it's started outside of Home Assistant

**Reply by jianyu-li** [reply]
It may have resolved itself. Today I did note that my apple tv went from OFF to PLAYING and home assistant was able to pick up on it 
<img width="561" height="316" alt="image" src="https://github.com/user-attachments/assets/e84bbebd-31a1-4b68-a7ce-68abfbb89a9b" />
